prism_compile.c

```diff
@@ -7164,9 +7164,16 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
             // is popped, then we know we don't need to do anything since it's
             // statically known.
             if (!popped) {
-                VALUE value = pm_static_literal_value(iseq, node, scope_node);
-                PUSH_INSN1(ret, location, duphash, value);
-                RB_OBJ_WRITTEN(iseq, Qundef, value);
+                const pm_hash_node_t *cast = (const pm_hash_node_t *) node;
+
+                if (&cast->elements.size == 0) {
```

Comment:
I think this is almost it, but this is checking if the address for the `size` member is 0, which I think effectively is checking if it's null. I think instead we want:

```suggestion
                if (cast->elements.size == 0) {
```